src/benchmarks/micro/corefx/System.Linq/Perf.OrderBy.cs

```diff
@@ -0,0 +1,151 @@
+"Quinn", "Blake", "Maxwell", "Pope", "Floyd", "Osborne", "Paul", "Mccarthy", "Guerrero", "Lindsey", "Estrada", "Sandoval", "Gibbs", "Tyler", "Gross", "Fitzgerald", "Stokes", "Doyle", "Sherman", "Saunders", "Wise",
+"Colon", "Gill", "Alvarado", "Greer", "Padilla", "Simon", "Waters", "Nunez", "Ballard", "Schwartz", "Mcbride", "Houston", "Christensen", "Klein", "Pratt", "Briggs", "Parsons", "Mclaughlin", "Zimmerman", "French",
+"Buchanan", "Moran", "Copeland", "Roy", "Pittman", "Brady", "Mccormick", "Holloway", "Brock", "Poole", "Frank", "Logan", "Owen", "Bass", "Marsh", "Drake", "Wong", "Jefferson", "Park", "Morton", "Abbott", "Sparks",
+"Patrick", "Norton", "Huff", "Clayton", "Massey", "Lloyd", "Figueroa", "Carson", "Bowers", "Roberson", "Barton", "Tran", "Lamb", "Harrington", "Casey", "Boone", "Cortez", "Clarke", "Mathis", "Singleton", "Wilkins",
+"Cain", "Bryan", "Underwood", "Hogan", "Mckenzie", "Collier", "Luna", "Phelps", "Mcguire", "Allison", "Bridges", "Wilkerson", "Nash", "Summers", "Atkins", "Wilcox", "Pitts", "Conley", "Marquez", "Burnett", "Richard",
+"Cochran", "Chase", "Davenport", "Hood", "Gates", "Clay", "Ayala", "Sawyer", "Roman", "Vazquez", "Dickerson", "Hodge", "Acosta", "Flynn", "Espinoza", "Nicholson", "Monroe", "Wolf", "Morrow", "Kirk", "Randall",
+"Anthony", "Whitaker", "Oconnor", "Skinner", "Ware", "Molina", "Kirby", "Huffman", "Bradford", "Charles", "Gilmore", "Dominguez", "Oneal", "Bruce", "Lang", "Combs", "Kramer", "Heath", "Hancock", "Gallagher", "Gaines",
+"Shaffer", "Short", "Wiggins", "Mathews", "Mcclain", "Fischer", "Wall", "Small", "Melton", "Hensley", "Bond", "Dyer", "Cameron", "Grimes", "Contreras", "Christian", "Wyatt", "Baxter", "Snow", "Mosley", "Shepherd",
+"Larsen", "Hoover", "Beasley", "Glenn", "Petersen", "Whitehead", "Meyers", "Keith", "Garrison", "Vincent", "Shields", "Horn", "Savage", "Olsen", "Schroeder", "Hartman", "Woodard", "Mueller", "Kemp", "Deleon",
+        };
+    }
+
+    [BenchmarkCategory(Categories.CoreFX, Categories.LINQ)]
+    public class Perf_OrderBy
+    {
+        private readonly Consumer _consumer = new Consumer();
+
+        [Params(0, 1, 10, 100, 1000)]
+        public int NumberOfPeople;
+
+        private Person[] People;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            People = PersonData.Generate(NumberOfPeople);
+        }
+
+        [Benchmark]
+        public void OrderByFirstName() => People.OrderBy(p => p.FirstName).Consume(_consumer);
+        [Benchmark]
+        public void OrderByFullName() => People.OrderBy(p => p.FullName).Consume(_consumer);
+        [Benchmark]
+        public void OrderByDOB() => People.OrderBy(p => p.DOB).Consume(_consumer);
+        [Benchmark]
+        public void OrderByFavouriteColour() => People.OrderBy(p => p.FavouriteColour, FavourColourComparer.Instance).Consume(_consumer);
```

Comment:
this is something we were lacking so far :+1:  but we need a better name

```suggestion
        public void OrderByCustomComparer() => People.OrderBy(p => p.FavouriteColour, FavourColourComparer.Instance).Consume(_consumer);
```